packages/cli/src/prompts/init-params.ts

```diff
@@ -1,29 +1,37 @@
+import { ContractMethodMutability as Mutability } from 'zos-lib';
 import pickBy from 'lodash.pickby';
 import isUndefined from 'lodash.isundefined';
 import negate from 'lodash.negate';
+
 import { parseInit } from '../utils/input';
 import { promptIfNeeded, argsList, InquirerQuestions } from './prompt';
 
 type PropsFn = (any) => InquirerQuestions;
 
-export default async function promptForInitParams(contractFullName: string, getCommandProps: PropsFn, options: any): Promise<{ initMethod: string, initArgs: string[] }> {
+export default async function promptForMethodParams(
+  contractFullName: string,
+  getCommandProps: PropsFn,
+  options: any,
+  additionalOpts: { [key: string]: string } = {},
+  constant: Mutability = Mutability.NotConstant
+): Promise<{ initMethod: string, initArgs: string[] }> {
+
   const { interactive } = options;
   let { initMethod, initArgs } = parseInit(options, 'initialize');
-  const { init: rawInitMethod } = options;
-  const opts = { askForInitParams: rawInitMethod, initMethod };
-  const initMethodProps = getCommandProps({ contractFullName, initMethod });
+  const opts = { ...additionalOpts, initMethod };
+  const methodProps = getCommandProps({ contractFullName, initMethod });
 
-  // prompt for init method if not provided
-  ({ initMethod } = await promptIfNeeded({ opts, props: initMethodProps }, interactive));
+  // prompt for method name if not provided
+  ({ initMethod } = await promptIfNeeded({ opts, props: methodProps }, interactive));
 
-  const initArgsKeys = argsList(contractFullName, initMethod.selector)
+  const methodArgsKeys = argsList(contractFullName, initMethod.selector, constant)
     .reduce((accum, current) => ({ ...accum, [current]: undefined }), {});
 
   // if there are no initArgs defined, or the initArgs array length provided is smaller than the
   // number of arguments in the function, prompt for remaining arguments
-  if (!initArgs || initArgs.length < Object.keys(initArgsKeys).length) {
-    const initArgsProps = getCommandProps({ contractFullName, initMethod: initMethod.selector, initArgs });
-    const promptedArgs = await promptIfNeeded({ opts: initArgsKeys, props: initArgsProps }, interactive);
+  if (!initArgs || initArgs.length < Object.keys(methodArgsKeys).length) {
+    const methodArgsProps = getCommandProps({ contractFullName, initMethod: initMethod.selector, initArgs });
+    const promptedArgs = await promptIfNeeded({ opts: methodArgsKeys, props: methodArgsProps }, interactive);
```

Comment:
Not sure I'm happy with the initArgs => methodArgs rename leaking to 2.3, but it seems to be contained to just this method.